packages/app-desktop/gui/NoteEditor/utils/resourceHandling.ts

```diff
@@ -125,3 +126,17 @@ export async function handlePasteEvent(event: any) {
 	}
 	return output;
 }
+
+export async function processPastedHtml(html: string) {
+	return await htmlUtils.replaceImageUrlsAsync(html, async (imageSrc: string) => {
+		if (imageSrc.startsWith('file')) {
+			const createdResource = await shim.createResourceFromPath(imageSrc.substr(7));
+			return `file://${Resource.fullPath(createdResource)}`;
+		} else {
+			const filePath = `${Setting.value('tempDir')}/${md5(Date.now())}`;
+			await shim.fetchBlob(imageSrc, { path: filePath });
+			const createdResource = await shim.createResourceFromPath(filePath);
+			return `file://${Resource.fullPath(createdResource)}`;
+		}
+	});
+}
```

Comment:
Rather than create a duplicate `replaceImageUrlsAsync` function, could you use the existing one `replaceImageUrls` in the following way:
- Loop through all the images using replaceImageUrls, and collect all the urls
- Process these `file://`, creating resources for each one and store the result in an associative array
- Run again replaceImageUrls, this time replacing the urls with the resource URL you've created

Also what happens if I paste some HTML text that contains a Joplin resource? According to your code it seems you would be duplicating the resource.

Reply:
Done. Also, modified the function to not create duplicate resources.